src/test/java/com/google/gcp_variant_transforms/library/HeaderIteratorTest.java

```diff
@@ -0,0 +1,145 @@
+// Copyright 2020 Google LLC
+
+package com.google.gcp_variant_transforms.library;
+
+import com.google.gcp_variant_transforms.library.HeaderIterator;
+import static com.google.common.truth.Truth.*;
+import com.google.common.collect.ImmutableList;
+import org.junit.Test;
+import org.junit.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * Units tests for HeaderIterator.java
+ */
+public class HeaderIteratorTest {
+
+  @Test
+  public void testHeaderLinesHasNext_whenCheckingBoolean_thenTrue() {
+    // Arrange
+    int listSize = 5;
+    ImmutableList.Builder<String> headerLinesBuilder = new ImmutableList.Builder<String>();
+    ImmutableList<String> headerLines;
+    HeaderIterator headerIterator;
+    for (int i = 0; i < listSize; i++) {
+      headerLinesBuilder.add("##sampleHeaderLine=" + i);
+    }
+    headerLines = headerLinesBuilder.build();
+    headerIterator = new HeaderIterator(headerLines);
+    // Act
+
+    // Assert
+    assertThat(headerIterator.hasNext()).isTrue();
+  }
+
+  @Test
+  public void testHeaderLinesHasNext_whenCheckingBoolean_thenFalse() {
+    // Arrange
+    int listSize = 5;
+    ImmutableList.Builder<String> headerLinesBuilder = new ImmutableList.Builder<String>();
+    ImmutableList<String> headerLines;
+    HeaderIterator headerIterator;
+    for (int i = 0; i < listSize; i++) {
+      headerLinesBuilder.add("##sampleHeaderLine=" + i);
+    }
+    headerLines = headerLinesBuilder.build();
+    headerIterator = new HeaderIterator(headerLines);
```

Comment:
Since headerIterator block is repeated in this and previous test, lets extract them into a common method.

Reply:
great idea! this really cleaned up the tests
resolved